src/tests/ftest/telemetry/wal_metrics.py

```diff
@@ -0,0 +1,217 @@
+"""
+  (C) Copyright 2018-2024 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+import time
+
+from ior_utils import write_data
+from telemetry_test_base import TestWithTelemetry
+from test_utils_pool import add_pool
+
+
+class WalMetrics(TestWithTelemetry):
+    """Tests for new specific metrics to track activity of md_on_ssd.
+
+    :avocado: recursive
+    """
+
+    def test_wal_commit_metrics(self):
+        """JIRA ID: DAOS-11626.
+
+        The WAL commit metrics is per-pool metrics, it includes 'wal_sz', 'wal_qd' and 'wal_waiters'
+        (see vos_metrics_alloc() in src/vos/vos_common.c). WAL commit metrics are updated on each
+        local transaction (for example, transaction for a update request, etc.)
+
+        Test steps:
+        1) Create a pool
+        2) Verify WAL commit metrics after pool creation (non-zero w/ MD on SSD)
+
+        :avocado: tags=all,daily_regression
+        :avocado: tags=hw,medium
+        :avocado: tags=telemetry
+        :avocado: tags=WalMetrics,test_wal_commit_metrics
+        """
+        wal_metrics = list(self.telemetry.ENGINE_POOL_VOS_WAL_METRICS)
+
+        self.log_step('Creating a pool (dmg pool create)')
+        add_pool(self)
+
+        self.log_step(
+            'Collect WAL commit metrics after creating a pool (dmg telemetry metrics query)')
+        ranges = self.telemetry.collect_data(wal_metrics)
+        for metric in list(ranges):
+            if '_sz' in metric and not metric.endswith('_mean') and not metric.endswith('_stddev'):
+                for label in ranges[metric]:
+                    if self.server_managers[0].manager.job.using_control_metadata:
+                        # The min/max/actual size should be greater than 0 for MD on SSD
+                        ranges[metric][label] = [1]
+                    else:
+                        ranges[metric][label] = [0, 0]
+            elif '_waiters' not in metric:
+                ranges.pop(metric)
+        if self.server_managers[0].manager.job.using_control_metadata:
+            self.log_step(
+                'Verify WAL commit size metrics are > 0 and waiters are 0 after creating a pool')
+        else:
+            self.log_step('Verify WAL commit metrics are 0 after creating a pool')
+        if not self.telemetry.verify_data(ranges):
+            self.fail('Unexpected WAL commit metric values after pool create')
+
+        self.log_step('Test passed')
+
+    def test_wal_reply_metrics(self):
+        """JIRA ID: DAOS-11626.
+
+        The WAL replay metrics is per-pool metrics in 'vos_rehydration' under each pool folder, it
```

Comment:
in 'vos_wal' instead of 'vos_rehydration'.

Reply:
Thanks, if I don't need to push again I'd like to fix it later if that's ok.